src/main/tools/daemonize.cc

```diff
@@ -324,7 +327,13 @@ static void MoveToCgroup(pid_t pid, const char* cgroup_path) {
         strcmp(fs_vfstype, "cgroup2") == 0) {
       char* procs_path;
       asprintf(&procs_path, "%s%s/cgroup.procs", fs_file, cgroup_path);
-      WriteFile(procs_path, "%d", pid);
+      FILE* procs = fopen(procs_path, "w");
+      if (procs != NULL) {
+        fprintf(procs, "%d", pid);
```

Comment:
Could you add checks and additional logging to `fprintf` and `fclose` similar to lines 94 & 97 to indicate that we failed to start the server in the specified cgroup?